ducktape/tests/loader.py

```diff
@@ -230,10 +250,24 @@ def expand_class(self, t_info):
                 t.function =_inject(**self.test_parameters)(t.function)
                 t.injected_args = self.test_parameters
                 test_info_list.append(t)
-
+                
+        test_info_list.extend(self.expand_regressions(test_info_list))
         return test_info_list
-
-    def expand_function(self, t_info):
+    
+    def expand_regressions(self, t_info_list):
+        regression_list = []
+        
+        for t_info in t_info_list:
+            if has_regression(t_info.function):
```

Comment:
Hmm, given the way this is used, it might be cleaner to provide a `regressions(t_info.function)` method that is a generator of `RegressionMark` annotations. As it is now, you scan through the list to figure out if there is a regression (`has_regression`), then scan through again doing the same check that `has_regression` does to filter down to only the `RegressionMark`s.